File: src/MissingInteger.java
import java.util.HashSet;

public class MissingInteger {

    public int solution(int[] A) {
        int num = 1;
        HashSet<Integer> hset = new HashSet<Integer>();

        for (int i = 0 ; i < A.length; i++) {
            hset.add(A[i]);
        }

        while (hset.contains(num)) {
            num++;
        }

        return num;
    }

    public static void main(String args[]) {
        MissingInteger MI = new MissingInteger();
        System.out.println(MI.solution(new int[]{-1, -3}));
    }
}

File: src/JS/MissingInteger.js
function MissingInteger(A) {
  let sets = new Set();
  let num = 1;

  for (let i = 0; i < A.length; i++) {
    sets.add(A[i]);
  }

  while (sets.has(num)) {
    num++;
  }
  return num;
}

console.log(MissingInteger([1, 3, 6, 5, 4, 2]));
